modules/api-import-export/src/main/java/org/wso2/carbon/apimgt/importexport/utils/APIImportUtil.java

```diff
@@ -368,6 +368,158 @@ public static void importAPI(String pathToArchive, String currentUser, boolean i
+     * @param isDefaultProviderAllowed Is default provider allowed to do the operation
+     * @throws APIImportException If there is an error the exception is thrown
+     */
+    public static void updateAPI(String apiID, String pathToArchive, String currentUser, boolean isDefaultProviderAllowed)
+            throws APIImportException {
+
+        API importedApi = null;
+        API targetApi = null;
+        String prevProvider;
+        APIDefinition definitionFromOpenAPISpec = new APIDefinitionFromOpenAPISpec();
+
+        String pathToJSONFile = pathToArchive + APIImportExportConstants.JSON_FILE_LOCATION;
+
+        try {
+            String jsonContent = FileUtils.readFileToString(new File(pathToJSONFile));
+            JsonElement configElement = new JsonParser().parse(jsonContent);
+            JsonObject configObject = configElement.getAsJsonObject();
+
+            //locate the "providerName" within the "id" and set it as the current user
+            JsonObject apiId = configObject.getAsJsonObject(APIImportExportConstants.ID_ELEMENT);
+            prevProvider = apiId.get(APIImportExportConstants.PROVIDER_ELEMENT).getAsString();
+            String prevTenantDomain = MultitenantUtils
+                    .getTenantDomain(APIUtil.replaceEmailDomainBack(prevProvider));
+            String currentTenantDomain = MultitenantUtils
+                    .getTenantDomain(APIUtil.replaceEmailDomainBack(currentUser));
+
+            // If the original provider is preserved,
+            if (isDefaultProviderAllowed) {
+                if (!StringUtils.equals(prevTenantDomain, currentTenantDomain)) {
+                    String errorMessage = "Tenant mismatch! Please enable preserveProvider property " +
+                            "for cross tenant API Import ";
+                    log.error(errorMessage);
+                    throw new APIImportException(errorMessage);
+                }
+                importedApi = new Gson().fromJson(configElement, API.class);
+            } else {
+                String prevProviderWithDomain = APIUtil.replaceEmailDomain(prevProvider);
+                String currentUserWithDomain = APIUtil.replaceEmailDomain(currentUser);
+                apiId.addProperty(APIImportExportConstants.PROVIDER_ELEMENT, currentUserWithDomain);
+                if (configObject.get(APIImportExportConstants.WSDL_URL) != null) {
+                    configObject.addProperty(APIImportExportConstants.WSDL_URL,
+                            configObject.get(APIImportExportConstants.WSDL_URL).getAsString()
+                                    .replace(prevProviderWithDomain, currentUserWithDomain));
+                }
+
+                importedApi = new Gson().fromJson(configElement, API.class);
+                //Replace context to match with current provider
+                importedApi = SetCurrentProvidertoAPIProperties(importedApi, currentTenantDomain, prevTenantDomain);
+            }
```

Comment:
there are multiple blank spaces in this class, please remove it and format.

Reply:
sure